design/ceph/multus-network.md

```diff
@@ -0,0 +1,110 @@
+- `cluster`: data daemon cluster network (binds to `cluster_network` Ceph option).
+
+If only `public` is set then `cluster` will take the value of `public`.
+
+## Multus supported configuration
+
+### Interface type
+
+As part of the CNI spec, Multus supports several [interface types](https://github.com/containernetworking/plugins#main-interface-creating).
+Rook will naturally support any of them as they don't fundamentally change the working behavior.
+
+### IPAM type
+
+This is where things get more complex.
+Currently there are [three different IPAM](https://github.com/containernetworking/plugins#ipam-ip-address-allocation) solutions available.
+
+As part of my research I have found that the following IPAM types are not good candidates:
+
+- host-local: Maintains a local database of allocated IPs, this only works on a per host basis, so not suitable for a distributed environment since we will end up with IP collision
+To fix this, the [whereabouts project](https://github.com/dougbtv/whereabouts) looks promising but is not officially supported.
+- static: Allocate a static IPv4/IPv6 addresses to container and it's useful in debugging purpose.
+This cannot at scale because this means we will have to allocate IPs for **all** the daemon so it's not scalable.
+
+## Monitors and OSDs implementation challenges
+
+The Ceph monitors only need access to the public network.
+The OSDs needs access to both public and cluster networks.
+
+Monitors requirements so far are the following:
+
+- Predictable IP addresses
+- Keep the same IP address for the life time of the pod (IP should survive a restart)
+
+### First proposal: IPAM type 'DHCP'
+
+In this scenario, a DHCP server will distribute an IP address to a pod using a given range.
+
+Pros:
+
+- Pods will get a dedicated IP on a physical network interface
+- No changes required in Ceph, Rook will detect the CIDR via the `NetworkAttachmentDefinition` then populate Ceph's flag `public_network` and `cluster_network`
+
+Cons:
+
+- IP allocation not predictable, we don't know it until the pod is up and running.
+So the detection must append inside the monitor container is running, similarly to what the OSD code does today.
+- Requires drastic changes in the monitor bootstrap code
+
+Assuming we go with this solution, we might need to change the way the monitors are bootstrapped:
+
+1. let the monitor discovered its own IP based on an interface
+2. once the first mon is bootstrapped, we register its IP in a ConfigMap as well as populating clusterInfo
+3. boot the second mon, look up in clusterInfo for the initial member (if the op dies in the process, we always `CreateOrLoadClusterInfo()` as each startup based on the cm so no worries)
+4. go on and on with the rest of the monitors
+
+TBT: if the pod restarts, it keeps the same IP.
+
+### Second proposal: IPAM type 'DHCP' with service IP
+
+We could and this is pure theory at this point use a IPAM with DHCP along with service IP.
```

Comment:
DHCP means IP addresses can change, which leads to the same question of how to deal with mon IPs changing...